rqt_image_view/src/rqt_image_view/image_view.cpp

```diff
@@ -77,11 +77,20 @@ void ImageView::initPlugin(qt_gui_cpp::PluginContext& context)
 
   ui_.save_as_image_push_button->setIcon(QIcon::fromTheme("image"));
   connect(ui_.save_as_image_push_button, SIGNAL(pressed()), this, SLOT(saveImage()));
+
+  pub_topic_custom_ = false;
```

Comment:
Actually I just saw why it was used. 

If you edit manually the line edit widget, the variable is set to true and what is in the line edit doesn't change even if you change topics. 
If you don't edit manually the line edit widget, when you change the subscription topic the publication topic will change accordingly.